txn_writer.go

```diff
@@ -162,20 +159,23 @@ func (diff StringDiff) MarshalJSON() ([]byte, error) {
 //   written out based on newLock if present, else lock, else error.
 func (sw *SafeWriter) Prepare(manifest *Manifest, lock *Lock, newLock *Lock, forceVendor bool) {
 	sw.Payload = &SafeWriterPayload{
-		Manifest:         manifest,
-		ForceWriteVendor: forceVendor,
+		Manifest:    manifest,
+		WriteVendor: forceVendor,
 	}
 
 	if newLock != nil {
 		if lock == nil {
 			sw.Payload.Lock = newLock
-			sw.Payload.ForceWriteVendor = true
+			sw.Payload.WriteVendor = true
 		} else {
 			diff := diffLocks(lock, newLock)
```

Comment:
Running `dep ensure -n` without a `lock.json` file does not print a diff even though `dep ensure` creates a new lock file. I don't think that's WAI?

With the other changes you made here this entire block in `newLock != nil` could then be simplified to:
```go
if newLock != nil {
    sw.Payload.Lock = newLock
    sw.Payload.LockDiff = diffLocks(lock, newLock)
    sw.Payload.WriteVendor = true
} ...
```
which solves the problem. However, you'd probably have to fix the [nil check](https://github.com/golang/dep/blob/55422bc13ba89f99c3889ccdca701799f2d1d6a0/txn_writer.go#L390) in the diffLocks function for this to work when lock is nil.